### PR TITLE
add phone number unit test

### PR DESCRIPTION
**Pull request by Fbasham**
Not associated with a card in ADO.  Add a unit test for the Phone Number component.

### CODE REVIEW

File: frontend/__tests__/components/phone-number.test.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { render, screen, waitFor } from '@testing-library/react';
+
+import { describe, expect, it } from 'vitest';
+
+import { PhoneNumber } from '~/components/phone-number';
+
+describe('<PhoneNumber />', () => {
+  it('Renders component with phone number', async () => {
+    render(<PhoneNumber phoneNumber="1234567890" />);
+    const element: HTMLInputElement = await waitFor(() => screen.findByTestId('phoneNumber'));
```
Comment (gregory-j-baker) [inline frontend/__tests__/components/phone-number.test.tsx:10]: Is the `waitfor(..)` required here? This seems to work but I'm not sure if it's working by accident:

``` typescript
const element = await screen.findByTestId<HTMLInputElement>('phoneNumber')
```

**Review by gregory-j-baker — APPROVED** [review]
Approving. Feel free to resolve my comment if it is nonsense.

File: frontend/__tests__/components/phone-number.test.tsx (hunk 1 above)
Comment (Fbasham) [reply frontend/__tests__/components/phone-number.test.tsx]: I guess not.  Anyways, removed from all instances.

File: frontend/__tests__/components/phone-number.test.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    const requiredLabel = await waitFor(() => screen.getAllByLabelText(/required/));
+    expect(requiredLabel).toBeDefined();
+    const element: HTMLInputElement = await screen.findByTestId('phoneNumber');
+    expect(element.disabled).toBe(false);
+  });
+
+  it('Renders component with field errors', async () => {
+    render(<PhoneNumber phoneNumber="1234567890" editMode fieldErrors={['testing-1-2-3', 'additional-error']} />);
+    let err = [...(document.querySelectorAll('.wb-server-error') || [])];
+    let fieldErrors = err?.map((x) => x.querySelector('.mrgn-lft-sm')?.textContent);
```
Comment (gregory-j-baker) [inline frontend/__tests__/components/phone-number.test.tsx:33]: **Nitpicks:**

- prefer `const` over `let`
- prefer `??` over `||`
- the `[]` isn't needed
  ``` typescript
  const err = [...document.querySelectorAll('NO_EXIST')];
  console.debug(err);
  gives → []
  ```

**Review by gregory-j-baker — DISMISSED** [review]
Feel free to just resolve my comment if you disagree (it's a nitpick so I don't have a strong opinion one way or another).

File: frontend/__tests__/components/phone-number.test.tsx (hunk 2 above)
Comment (Fbasham) [reply frontend/__tests__/components/phone-number.test.tsx]: updated.  Anything else, boss?

**Review by gregory-j-baker — APPROVED** [review]

**Comment by gregory-j-baker** [comment]
FYI I rebased your branch on main so it would merge.

**Review by gregory-j-baker — APPROVED** [review]

**Pull request merged**